File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/game/PlayerTurnEndedCommand.java
package bluffinmuffin.protocol.commands.game;

import java.util.StringTokenizer;

import bluffinmuffin.poker.entities.type.PlayerActionType;
import bluffinmuffin.protocol.commands.AbstractCommand;

public class PlayerTurnEndedCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return PlayerTurnEndedCommand.COMMAND_NAME;
    }
    
    private final int m_playerPos;
    private final int m_playerBet;
    private final int m_playerMoney;
    private final int m_totalPot;
    private final PlayerActionType m_actionType;
    private final int m_actionAmount;
    private final boolean m_isPlaying;
    
    public static String COMMAND_NAME = "gamePLAYER_TURN_ENDED";
    
    public PlayerTurnEndedCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
        m_playerBet = Integer.parseInt(argsToken.nextToken());
        m_playerMoney = Integer.parseInt(argsToken.nextToken());
        m_totalPot = Integer.parseInt(argsToken.nextToken());
        m_actionType = PlayerActionType.values()[Integer.parseInt(argsToken.nextToken())];
        m_actionAmount = Integer.parseInt(argsToken.nextToken());
        m_isPlaying = Boolean.parseBoolean(argsToken.nextToken());
    }
    
    public PlayerTurnEndedCommand(int pos, int bet, int money, int totalPot, PlayerActionType actionType, int actionAmount, boolean isPlaying)
    {
        m_playerPos = pos;
        m_playerBet = bet;
        m_playerMoney = money;
        m_totalPot = totalPot;
        m_actionType = actionType;
        m_actionAmount = actionAmount;
        m_isPlaying = isPlaying;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerPos);
        append(sb, m_playerBet);
        append(sb, m_playerMoney);
        append(sb, m_totalPot);
        append(sb, m_actionType.ordinal());
        append(sb, m_actionAmount);
        append(sb, m_isPlaying);
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
    
    public int getPlayerBet()
    {
        return m_playerBet;
    }
    
    public int getPlayerMoney()
    {
        return m_playerMoney;
    }
    
    public int getTotalPot()
    {
        return m_totalPot;
    }
    
    public PlayerActionType getActionType()
    {
        return m_actionType;
    }
    
    public int getActionAmount()
    {
        return m_actionAmount;
    }
    
    public boolean isPlaying()
    {
        return m_isPlaying;
    }
}

File: trunk/C#/PokerProtocol/Commands/Game/PlayerTurnEndedCommand.cs
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;
using PokerWorld.Game;
using PokerWorld.Game.Enums;

namespace PokerProtocol.Commands.Game
{
    public class PlayerTurnEndedCommand : AbstractTextCommand
    {
        public static string COMMAND_NAME = "gamePLAYER_TURN_ENDED";

        private readonly int m_PlayerPos;
        private readonly int m_PlayerBet;
        private readonly int m_PlayerMoney;
        private readonly int m_TotalPot;
        private readonly GameActionEnum m_ActionType;
        private readonly int m_ActionAmount;
        private readonly bool m_IsPlaying;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int PlayerBet
        {
            get { return m_PlayerBet; }
        }
        public int PlayerMoney
        {
            get { return m_PlayerMoney; }
        }
        public int TotalPot
        {
            get { return m_TotalPot; }
        }
        public GameActionEnum ActionType
        {
            get { return m_ActionType; }
        }
        public int ActionAmount
        {
            get { return m_ActionAmount; }
        }
        public bool IsPlaying
        {
            get { return m_IsPlaying; }
        }

        public PlayerTurnEndedCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_PlayerBet = int.Parse(argsToken.NextToken());
            m_PlayerMoney = int.Parse(argsToken.NextToken());
            m_TotalPot = int.Parse(argsToken.NextToken());
            m_ActionType = (GameActionEnum)int.Parse(argsToken.NextToken());
            m_ActionAmount = int.Parse(argsToken.NextToken());
            m_IsPlaying = bool.Parse(argsToken.NextToken());
        }

        public PlayerTurnEndedCommand(int pos, int bet, int money, int totalPot, GameActionEnum actionType, int actionAmount, bool isPlaying)
        {
            m_PlayerPos = pos;
            m_PlayerBet = bet;
            m_PlayerMoney = money;
            m_TotalPot = totalPot;
            m_ActionType = actionType;
            m_ActionAmount = actionAmount;
            m_IsPlaying = isPlaying;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_PlayerBet);
            Append(sb, m_PlayerMoney);
            Append(sb, m_TotalPot);
            Append(sb, (int)m_ActionType);
            Append(sb, m_ActionAmount);
            Append(sb, m_IsPlaying);
        }
    }
}
